bcda/auth/api.go

```diff
@@ -26,28 +28,39 @@ import (
 		401: invalidCredentials
 		500: serverError
 */
+
 func GetAuthToken(w http.ResponseWriter, r *http.Request) {
 	clientId, secret, ok := r.BasicAuth()
 	if !ok {
 		http.Error(w, http.StatusText(http.StatusBadRequest), http.StatusBadRequest)
 		return
 	}
 
-	token, expiresIn, err := GetProvider().MakeAccessToken(Credentials{ClientID: clientId, ClientSecret: secret})
+	tokenInfo, err := GetProvider().MakeAccessToken(Credentials{ClientID: clientId, ClientSecret: secret})
 	if err != nil {
-		http.Error(w, http.StatusText(http.StatusUnauthorized), http.StatusUnauthorized)
+		switch err.(type) {
+		case *customErrors.RequestTimeoutError:
+			//default retrySeconds: 1 second (may convert to environmental variable later)
+			retrySeconds := strconv.FormatInt(int64(1), 10)
+			w.Header().Set("Retry-After", retrySeconds)
+			http.Error(w, err.Error(), http.StatusServiceUnavailable)
+		case *customErrors.UnexpectedSSASError:
+			http.Error(w, err.Error(), http.StatusInternalServerError)
+		default:
+			http.Error(w, err.Error(), http.StatusUnauthorized)
+		}
 		return
 	}
 
 	// https://tools.ietf.org/html/rfc6749#section-5.1
 
-	body := []byte(fmt.Sprintf(`{"access_token": "%s", "expires_in": "%s", "token_type":"bearer"}`, token, expiresIn))
 	w.Header().Set("Content-Type", "application/json")
 	w.Header().Set("Cache-Control", "no-store")
 	w.Header().Set("Pragma", "no-cache")
-	_, err = w.Write(body)
+	_, err = w.Write([]byte(tokenInfo))
```

Comment:
Is tokenInfo in same structure as before for the end user? 

{"access_token": "%s", "expires_in": "%s", "token_type":"bearer"}`

Reply:
Yes.